main.py

```diff
@@ -0,0 +1,63 @@
+import config
+from CanvasAPI import *
+
+
+def getCanvasInstance():
+    return CanvasAPI(config.Canvas.API_BASE_URL, authZToken=config.Canvas.API_AUTHZ_TOKEN)
+
+
+def getCourseIDsWithOutcome(canvas, courseIDs, outcome):
+    matchingCourseIDs = set()
+    for courseID in courseIDs:
+        courseOutcomeGroupLinks = canvas.getCoursesOutcomeGroupLinksObjects(courseID)
+
+        # Is it possible to short-circuit this using itertools?
+        matchingCourseIDs.update(
+            {courseID for outcomeLink in courseOutcomeGroupLinks
+             if outcomeLink.outcome.id == outcome.id}
+        )
+    return matchingCourseIDs
+
+
+def getCourseAssignmentsWithOutcome(canvas, courseIDs, outcome):
```

Comment:
Just a thought:  Recoding this using lambdas and map and filter would be an interesting exercise.  That ought to get rid of a lot of the boilerplate code of for/in, if, break.